graphql/parser.go

```diff
@@ -138,13 +144,22 @@ func parseSelectionSet(input *ast.SelectionSet, globalFragments map[string]*Frag
 				return nil, NewClientError("unknown fragment")
 			}
 
-			fragments = append(fragments, fragment)
+			directives, err := parseDirectives(selection.Directives, vars)
+			if err != nil {
+				return nil, err
+			}
+
+			fragmentCopy := *fragment
```

Comment:
alternatively, we could create a `FragmentInstance` type, but it seems ~equivalent.